src/test/ui/imports/issue-55884-2.stderr

```diff
@@ -1,8 +1,20 @@
 error[E0603]: struct `ParseOptions` is private
   --> $DIR/issue-55884-2.rs:12:17
    |
+LL |     pub struct ParseOptions {}
+   |     -------------------------- consider importing struct `::options::ParseOptions` directly
+...
+LL |     pub use options::*;
+   |             ---------- ...through this public re-export of `ParseOptions`
+...
+LL |     use ParseOptions;
+   |         ------------ used restricted re-export of `ParseOptions`...
```

Comment:
I don't understand what this line is saying: is it meant to be a continuation of the previous two suggestions?

Reply:
It seems from the grammar that it isn't, but that's how I would read it at first (especially with the nice flowing diagnostics around lifetime management).

Reply:
`pub use parser::ParseOptions;` is importing `pub use options::*;`, which imports this `use ParseOptions;`, which imports the `struct`. I am _not_ fully clear on what is going on with the "...through this public re-export of `ParseOptions`..." label from below, but that is exactly what this test is testing: bad resolution due to a shadowing glob import.

Would this be better for readability if this were using `span_note`s for the ordering?

```
error[E0603]: struct `ParseOptions` is private
  --> $DIR/issue-55884-2.rs:12:17
   |
LL |     pub struct ParseOptions {}
   |                ------------ consider importing struct `::options::ParseOptions` directly
...
LL | pub use parser::ParseOptions; //~ ERROR struct `ParseOptions` is private
   |                 ^^^^^^^^^^^^ this struct is private
note: used restricted re-export of `ParseOptions`...
   |
LL |     use ParseOptions;
   |         ^^^^^^^^^^^^
note: ...through this public re-export of `ParseOptions`...
   |
LL | pub use parser::ParseOptions; //~ ERROR struct `ParseOptions` is private
   |         ^^^^^^^^^^^^^^^^^^^^
note: ...through this public re-export of `ParseOptions`
   |
LL |     pub use options::*;
   |             ^^^^^^^^^^
```

Reply:
latest commit does this